resolwe/flow/models.py

```diff
@@ -349,6 +351,35 @@ def __init__(self, *args, **kwargs):
         super(Data, self).__init__(*args, **kwargs)
         self._original_name = self.name
 
+    def save_storage(self, instance, schema):
+        """Save basic:json values to a Storage collection."""
+
+        validator = {val['name']: val for val in schema}
+        for key in instance:
```

Comment:
You should use `iterate_fields` to avoid the duplication of the iteration logic.